src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,108 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_DEFAULT_LOGIN = "user@gmail.com";
+    private static final String VALID_DEFAULT_PASSWORD = "user1234";
+    private static final int VALID_AGE = 25;
+    private static final String INVALID_LOGIN_LENGTH = "user";
+    private static final String INVALID_LOGIN_START_WITH_NUMBERS = "1234@gmail.com";
+    private static final String INVALID_EMPTY_LOGIN = "";
+    private static final String INVALID_PASSWORD_LENGTH = "pass";
+    private static final String INVALID_EMPTY_PASSWORD = "";
+    private static final int INVALID_USER_AGE = 16;
+
+    private RegistrationService registrationService;
+    private User user;
+    
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+        user.setAge(VALID_AGE);
+        user.setLogin(VALID_DEFAULT_LOGIN);
+        user.setPassword(VALID_DEFAULT_PASSWORD);
+    }
+
+    @Test
+    void register_nullValueForUser_notOk() {
+        Assertions.assertThrows(RegistrationException.class, 
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_validUser_ok() {
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_invalidLoginLength_notOk() {
+        user.setLogin(INVALID_LOGIN_LENGTH);
+        Assertions.assertThrows(RegistrationException.class, 
+                () -> registrationService.register(user));
+    }

```

Comment:
Here, you probably want to check if the exception message that you receive is the one that you expect.
```suggestion
   Exception exception = assertThrows(
                RegistrationException.class,
                () -> registrationService.register(user));
        String actual = exception.getMessage();
        assertEquals(expected, actual);```
I'm not sure if it's necessary, but I think it's a good practice to ensure that you're throwing the proper exception from the proper place.